cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
--   **DO NOT** use X-Content-Security-Policy or X-WebKit-CSP. Their implementations are obsolete (since Firefox 23, Chrome 25), limited, inconsistent, and incredibly buggy.
+Multiple types of directives exist that allow the developer to granularly control the flow of the policies.
 
-Directives
-----------
+## Fetch Directives
 
-The following is a listing of directives, and a brief description.
+Fetch directives tell the browser the locations to trust and load resources from.
 
-### CSP 1.0 Spec
+- `default-src` is a fallback directive for the other fetch directives. Directives that are specified have no inheritance, yet directives that are not specified will fall back to the value of `default-src`.
+- `child-src` allows the developer to control nested browsing contexts and worker execution contexts.
+  - `frame-src` specifies the URLs which can be loaded into nested browsing contexts (*e.g.* `<iframe>`).
+  - `worker-src` specifies the URLs which can be loaded as worker, sharedworker, or serviceworker. Fallback's on `script-src` too.
+- `connect-src` provides control over fetch requests, XHR, eventsource, beacon and websockets connections.
+- `font-src` specifies which URLs to load fonts from.
 
--   **connect-src** (d) - restricts which URLs the protected resource can load using script interfaces. (e.g. send() method of an XMLHttpRequest object)
--   **font-src** (d) - restricts from where the protected resource can load fonts
--   **img-src** (d) - restricts from where the protected resource can load images
--   **media-src** (d) - restricts from where the protected resource can load video, audio, and associated text tracks
--   **object-src** (d) - restricts from where the protected resource can load plugins
--   **script-src** (d) - restricts which scripts the protected resource can execute. Additional restrictions against, inline scripts, and eval. Additional directives in CSP2 for hash and nonce support
--   **style-src** (d) - restricts which styles the user may applies to the protected resource. Additional restrictions against inline and eval.
--   **default-src** - Covers any directive with *(d)*
--   **frame-src** - restricts from where the protected resource can embed frames. Note, deprecated in CSP2
--   **report-uri** - specifies a URL to which the user agent sends reports about policy violation
--   **sandbox** - specifies an HTML sandbox policy that the user agent applies to the protected resource. Optional in 1.0
+- `img-src` specifies the URLs that images can be loaded from.
+- `manifest-src` specifies the URLs that application manifests may be loaded from.
+- `media-src` specifies the URLs from which video, audio and text track resources can be loaded from.
+- `prefetch-src` specifies the URLs from which resources can be prefetched from.
+- `object-src` specifies the URLs from which plugins can be loaded from.
```

Comment:
According to [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/object-src), it is the directive *[plugin-types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/plugin-types)* that instruct which plugins are allowed.

Reply:
Following the specs from [w3c](https://www.w3.org/TR/CSP3/#directive-object-src) and the MDN link you mentioned, `plugin-types` specify the the allowed types, and not the sources from which they will be loaded.

Reply:
Yep, my mistake, i have mingled the type and the source, sorry.